pkg/jobconfig/files.go

```diff
@@ -499,3 +474,81 @@ func MakeRegexFilenameLabel(possibleRegex string) string {
 	}
 	return label
 }
+
+func isStale(job prowconfig.JobBase, newLabel ProwgenLabel) bool {
+	genLabel, generated := job.Labels[ProwJobLabelGenerated]
+	return generated && genLabel != string(newLabel)
+}
+
+func isGenerated(job prowconfig.JobBase) bool {
+	_, generated := job.Labels[ProwJobLabelGenerated]
+	return generated
+}
+
+func PruneStaleJobs(jobDir string, oldLabel, newLabel ProwgenLabel) error {
+	if err := OperateOnJobConfigDir(jobDir, func(jobConfig *prowconfig.JobConfig, info *Info) error {
+		if info.Type == "periodics" {
```

Comment:
Technically this isn't necessary right? We do nothing without the labels

Reply:
It's not necessary, but it causes a read-and-write on periodics, which makes noise in the repo (periodics are handcrafted, so it fixes ordering, removes omitempty fields...).

Reply:
ah